lib/src/widgets/platform_scaffold.dart

```diff
@@ -0,0 +1,177 @@
+  }
+
+  CupertinoNavigationBar _iosBuilder(BuildContext context) {
+    return CupertinoNavigationBar(
+      key: key,
+      backgroundColor: Styles.cupertinoScaffoldColor.resolveFrom(context),
+      border: null,
+      padding: cupertinoPadding ?? Styles.cupertinoAppBarTrailingWidgetPadding,
+      middle: title,
+      trailing: Row(
+        mainAxisSize: MainAxisSize.min,
+        children: actions,
+      ),
+    );
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return PlatformWidget(
+      androidBuilder: _androidBuilder,
+      iosBuilder: _iosBuilder,
+    );
+  }
+}
+
+/// A platform-aware circular loading indicator to be used in [PlatformAppBar.actions].
+class PlatformAppBarLoadingIndicator extends StatelessWidget {
+  const PlatformAppBarLoadingIndicator({super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    return PlatformWidget(
+      iosBuilder: (_) => const CircularProgressIndicator.adaptive(),
+      androidBuilder: (_) => const Padding(
+        padding: EdgeInsets.only(right: 16),
+        child: SizedBox(
+          height: 24,
+          width: 24,
+          child: Center(
+            child: CircularProgressIndicator.adaptive(),
+          ),
+        ),
+      ),
+    );
+  }
+}
+
+class _CupertinoNavBarWrapper extends StatelessWidget
+    implements ObstructingPreferredSizeWidget {
+  const _CupertinoNavBarWrapper({
+    required this.child,
+  });
+
+  final Widget child;
+
+  @override
+  Widget build(BuildContext context) => child;
+
+  @override
+  Size get preferredSize =>
+      const Size.fromHeight(kMinInteractiveDimensionCupertino);
+
+  /// True if the navigation bar's background color has no transparency.
+  @override
+  bool shouldFullyObstruct(BuildContext context) {
+    final Color backgroundColor = CupertinoTheme.of(context).barBackgroundColor;
+    return backgroundColor.alpha == 0xFF;
+  }
+}
+
+/// A screen with a navigation bar and a body that adapts to the platform.
+///
+/// On Android, this is a [Scaffold] with an [AppBar],
+/// on iOS a [CupertinoPageScaffold] with a [CupertinoNavigationBar].
+///
+/// See [PlatformAppBar] for an app bar that adapts to the platform and needs to be passed to this widget.
+class PlatformScaffold extends StatelessWidget {
+  const PlatformScaffold({
+    super.key,
+    required this.appBar,
+    required this.body,
+    this.resizeToAvoidBottomInset = true,
+  });
+
+  /// Acts as the [AppBar] for Android and as the [CupertinoNavigationBar] for iOS.
+  ///
+  /// Usually an instance of [PlatformAppBar].
+  final Widget appBar;
+
+  /// The main content of the screen, displayed below the navigation bar.
+  final Widget body;
+
+  /// See [Scaffold.resizeToAvoidBottomInset] and [CupertinoPageScaffold.resizeToAvoidBottomInset]
+  final bool resizeToAvoidBottomInset;
+
+  Widget _androidBuilder(BuildContext context) {
+    return Scaffold(
+      resizeToAvoidBottomInset: resizeToAvoidBottomInset,
+      appBar: PreferredSize(
```

Comment:
Instead of wrapping with `PreferredSize`, would it be possible to have `PlatformAppBar` implements `ObstructingPreferredSizeWidget` and `PreferredSizeWidget`?

Reply:
I thought this was not possible because Android and iOS need different numeric values here, and we can't use `Theme.of(context).platform` because the `context` is not available.